junit-platform-engine/src/main/java/org/junit/platform/engine/EngineDiscoveryListener.java

```diff
@@ -53,4 +53,7 @@ public interface EngineDiscoveryListener {
 	default void selectorProcessed(UniqueId engineId, DiscoverySelector selector, SelectorResolutionResult result) {
 	}
 
+	default void issueFound(UniqueId engineId, EngineDiscoveryIssue issue) {
```

Comment:
:speech_balloon: This is the top-level API for engines to report discovery issues.